server/storage/backend/backend.go

```diff
@@ -199,6 +200,10 @@ func newBackend(bcfg BackendConfig) *backend {
 		bcfg.Logger.Panic("failed to open database", zap.String("path", bcfg.Path), zap.Error(err))
 	}
 
+	if bcfg.Logger == nil {
```

Comment:
I don't think it works. Both the `newBoltLoggerZap` call and Line 200 `panic` already use `bcfg.Logger`.
That assignment should be placed at the beginning of the `newBackend`

Reply:
Good catch!